cmd/clair/main.go

```diff
@@ -117,44 +117,26 @@ func main() {
 	})
 
 	// Signal handler goroutine.
-	srvs.Go(func() error {
-		c := make(chan os.Signal, 1)
-		signal.Notify(c, os.Interrupt)
+	go func() {
+		ctx, stop := signal.NotifyContext(ctx, os.Interrupt)
 		defer func() {
-			signal.Stop(c)
-			close(c)
-			zlog.Info(ctx).Msg("unregistered signal handler")
-		}()
-		zlog.Info(ctx).Msg("registered signal handler")
-		select {
-		case sig := <-c:
-			zlog.Info(ctx).
-				Stringer("signal", sig).
-				Msg("gracefully shutting down")
 			// Note that we're using the root context here, so that we get a
 			// full timeout if one errgroup goroutine returns uncleanly.
 			tctx, done := context.WithTimeout(ctx, 10*time.Second)
```

Comment:
```suggestion
			// Note that we're using a background context here, so that we get a
			// full timeout if the signal handler has fired.
			tctx, done := context.WithTimeout(context.Background(), 10*time.Second)
```